ikarus/controlroutines/controlstate.hh

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file controlstate.hh
+ * \brief Defines the ControlState structure for storing results of a control routine.
+ * \ingroup  controlroutines
+ */
+
+#pragma once
+
+#include <ikarus/solver/nonlinearsolver/solverstate.hh>
+
+namespace Ikarus {
+
+/**
+ * \struct ControlState
+ * \brief Structure containing information about the state of the control routine.
+ *
+ * \details This structure holds information about the success of a control routine,
+ * the flag indicating if the state of deformation is in the initial configuration,
+ * including current load step number, total number of iterations performed by the non-linear solver,
+ * load step size, value of the load factor, the solution vector,
+ * state of the non-linear solver and the name of the control routine.
+ *
+ * \ingroup  controlroutines
+ */
+struct ControlState
+{
+  bool success{false};      ///< Flag indicating the success of the control.
+  bool initialConfig{true}; ///< Flag indicating if the state of deformation is in the initial configuration
```

Comment:
To have this flag here, does not make sense.

Reply:
I included this such that a user can check and log information accordingly [(example)](https://github.com/ikarus-project/ikarus/blob/81614f52be2e03f3663d579826dd68789679bf11/ikarus/utils/observer/controllogger.cpp#L31) and also perhaps it tells if the state of the control routine is in the initial configuration or not. 

May be instead of calling [`updateAndNotifyControlState(controlState, nonOp, solverState);`](https://github.com/ikarus-project/ikarus/blob/81614f52be2e03f3663d579826dd68789679bf11/ikarus/controlroutines/pathfollowing.inl#L46) here, once can just update the control state with information from the non-linear operator and just notify `SOLUTION_CHANGED` and remove `STEP_ENDED` (as you once suggested). Then this would get rid of the requirement of having this `bool initialConfig`.

Reply:
So something like this?
For example, in `ikarus/controlroutines/pathfollowing.inl` we have
```cpp
ControlState controlState{
      .success = false, .currentStep = 0, .totalIterations = 0, .stepSize = stepSize_, .name = this->name()};
  auto& nonOp = nonLinearSolver_->nonLinearOperator();
 controlState.sol    = &nonOp.firstParameter();
 controlState.lambda = nonOp.lastParameter();
 this->notify(ControlMessages::CONTROL_STARTED, controlState);
```

and then in `ikarus/utils/observer/controlvtkwriter.hh` we just have 
```cpp
case ControlMessages::CONTROL_STARTED:
case ControlMessages::SOLUTION_CHANGED: { ... } break;
```